shared/domains/data/index.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+// =======================
+// Domain: Data persistence
+// =======================
+
+import _ from 'lodash'
+
+const _store = {}
+
+export default {
+  '/get': function (path) {
+    return _.get(_store, path, [])
+  },
+  '/set': function (path, data) {
+    _.set(_store, path, data)
+  },
+  '/add': function (path, data) {
+    if (_.get(_store, path)) {
+      _.get(_store, path).push(data)
+    } else {
+      _.set(_store, path, [data])
+    }
+  }
```

Comment:
Also, to avoid filling up memory and slowing down the computer, shouldn't there be corresponding `remove` type selectors here?

Reply:
Note:

- For `/set`, if you use a `WeakMap` then I think that means the objects will be automatically deallocated if they have no references outside of the map
- For `/add`, I don't think there's a `WeakArray`, so it might help to really think about how the lifecycle of data will be managed here to avoid bloating memory

Reply:
this is exactly why I gave the data handling its own domain, I'm sure we will come up with many better data management solutions in the near future, and this way we only have to change it here.